codice/ddf
# DDF-4139 DDF should validate the Subject sent in the AuthnRequest

#### What does this PR do?
Verifies that the Subject in the received AuthnRequest refers to the same principle (user) as the Subject in the created Response.

#### Who is reviewing it? 
@blen-desta 
@brjeter 

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@coyotesqrl

#### How should this be tested?
Run [this branch of the SAML CTK](https://github.com/codice/saml-conformance/pull/200) against the built & installed DDF. The invalid subject error tests should pass.

#### What are the relevant tickets?
[DDF-4139](https://codice.atlassian.net/browse/DDF-4139)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
